cmd/util/flags/cliflags/job.go

```diff
@@ -0,0 +1,232 @@
+	return emojiString
+}
+
+func SafeAnnotationRegex() *regexp.Regexp {
+	r := regexp.MustCompile(fmt.Sprintf("[^%s|^%s]", returnAllEmojiString(), RegexString))
+	return r
+}
+
+func (j *JobSettings) Labels() (map[string]string, error) {
+	rawLabels := j.labels
+	if j.cmd.Flags().Changed("labels") {
+		rawLabels = make(map[string]string)
+		for _, l := range j.legacy.labels {
+			// NB: the legacy flag is a string slice, but the task expects a map of key,value labels.
+			// We set only the key here.
+			rawLabels[l] = ""
+		}
+	}
+	for k, v := range rawLabels {
+		if !IsSafeAnnotation(k) || k == "" {
+			return nil, fmt.Errorf("label with key %q is not allowed", k)
+		}
+		if !IsSafeAnnotation(v) || v == "" {
+			return nil, fmt.Errorf("label with value %q is not allowed", k)
+		}
+	}
+
+	return rawLabels, nil
+}
+
+type LegacyJobFlags struct {
+	// Deprecated: use `JobSettings.jobType`
+	targetingMode model.TargetingMode
+	// Deprecated: use 'JobSettings.labels'
+	labels []string
+	// Deprecated: use 'JobSettings.constraints'
+	selectors string
+	//Deprecated: use `JobSettings.count`
+	concurrency int
+}
+
+func DefaultJobSettings() *JobSettings {
+	return &JobSettings{
+		name:        "main",
+		namespace:   "default",
```

Comment:
use `models.DefaultNamespace`

Reply:
Disregard previous comment, I was using the wrong validate method.